### PR TITLE
Turn off avahi and avahi-autoipd fixes during install

### LINKED ISSUE #10277: RHEL9 GUI Kickstart install of CIS Workstation Level  2 aborts
#### Description of problem:
Installation of a RHEL9 GUI via Kickstart with CIS Worksation Level 2 fails.
The Anaconda logs seem to indicate that the mountpoints don't exist.

#### SCAP Security Guide Version:
Upstream as of 2ce85d7

#### Operating System Version:
RHEL9

#### Steps to Reproduce:

Example install command
1. virt-install --name=test_suite_vm             --memory=4096 --vcpus=2 --hvm             --network bridge=virbr0 --disk path=<disk path>,size=20             --location <compose url> --initrd-inject ssg-rhel9-cis_workstation_l2-ks.cfg --wait 0             --graphics vnc --extra-args "inst.ks=file:/ssg-rhel9-cis_workstation_l2-ks.cfg inst.addrepo=AppStream,<appstream compose url> inst.addrepo=ssg,http://192.168.122.1/ssg console=ttyS0 inst.graphical notmux systemd.journald.forward_to_console=1" --noautoconsole'

#### Actual Results:
```
[  160.705959] anaconda[1803]: anaconda: misc: OSCAP Addon: Done with analysis
[  160.723837] org.fedoraproject.Anaconda.Modules.Storage[1858]: INFO:blivet:registered action: [157] create format xfs filesystem mounted at /var/tmp on lvmlv VolGroup-LogVol7 (id 151)
[  160.762998] org.fedoraproject.Anaconda.Modules.Storage[1858]: DEBUG:blivet:             DeviceTree.get_device_by_name: name: VolGroup ; incomplete: False ; hidden: False ;
[  160.799797] org.fedoraproject.Anaconda.Modules.Storage[1858]: DEBUG:blivet:             DeviceTree.get_device_by_name returned non-existent 19.5 GiB lvmvg VolGroup (118)
[  160.832353] org.fedoraproject.Anaconda.Modules.Storage[1858]: DEBUG:blivet:             DeviceTree.get_device_by_name: name: VolGroup-LogVol03 ; incomplete: False ; hidden: False ;
[  160.868332] org.fedoraproject.Anaconda.Modules.Storage[1858]: DEBUG:blivet:             DeviceTree.get_device_by_name returned None
[  160.896791] org.fedoraproject.Anaconda.Modules.Storage[1858]: DEBUG:blivet:                XFS.supported: supported: True ;
[  160.921923] org.fedoraproject.Anaconda.Modules.Storage[1858]: DEBUG:blivet:get_format('xfs') returning XFS instance with object id 159
[  160.949959] org.fedoraproject.Anaconda.Modules.Storage[1858]: DEBUG:blivet:get_format('None') returning DeviceFormat instance with object id 161
[  160.980221] org.fedoraproject.Anaconda.Modules.Storage[1858]: DEBUG:blivet:                              LVMVolumeGroupDevice.add_child: name: VolGroup ; child: LogVol03 ; kids: 4 ;
[  161.014676] org.fedoraproject.Anaconda.Modules.Storage[1858]: DEBUG:blivet:                            LVMLogicalVolumeDevice._set_format: VolGroup-LogVol03 ; type: xfs ; current: None ;
[  161.044840] org.fedoraproject.Anaconda.Modules.Storage[1858]: DEBUG:blivet:get_format('None') returning DeviceFormat instance with object id 163
[  161.068518] org.fedoraproject.Anaconda.Modules.Storage[1858]: DEBUG:blivet:                              LVMVolumeGroupDevice.remove_child: name: VolGroup ; child: LogVol03 ; kids: 5 ;
[  161.099594] org.fedoraproject.Anaconda.Modules.Storage[1858]: DEBUG:blivet:                            LVMVolumeGroupDevice.add_child: name: VolGroup ; child: LogVol03 ; kids: 4 ;
[  161.131051] org.fedoraproject.Anaconda.Modules.Storage[1858]: DEBUG:blivet:                          LVMLogicalVolumeDevice._set_format: VolGroup-LogVol03 ; type: xfs ; current: None ;
[  161.161780] org.fedoraproject.Anaconda.Modules.Storage[1858]: DEBUG:blivet:                            LVMLogicalVolumeDevice.read_current_size: exists: False ; path: /dev/mapper/VolGroup-LogVol03 ; sysfs_path:  ;
[  161.198133] org.fedoraproject.Anaconda.Modules.Storage[1858]: DEBUG:blivet:VolGroup size is 19.5 GiB

...[snip]...

[  171.403394] anaconda[1803]: anaconda: core.util: Skipping detection of SMT.
[  171.417759] anaconda[1803]: anaconda: misc: OSCAP addon: Executing subprocess: 'oscap xccdf generate fix --template=urn:redhat:anaconda:pre --profile=xccdf_org.ssgproject.content_profile_cis_workstation_l2 --datastream-id=scap_org.open-scap_datastream_from_xccdf_ssg-rhel9-xccdf.xml --xccdf-id=scap_org.open-scap_cref_ssg-rhel9-xccdf.xml /tmp/openscap_data/ssg-rhel9-ds.xml'
[  171.479330] org.fedoraproject.Anaconda.Modules.Storage[1858]: DEBUG:blivet:             DeviceTree.get_device_by_name: name: VolGroup-LogVol02 ; incomplete: True ; hidden: True ;
[  171.509148] org.fedoraproject.Anaconda.Modules.Storage[1858]: DEBUG:blivet:             DeviceTree.get_device_by_name returned non-existent 1024 MiB lvmlv VolGroup-LogVol02 (133) with non-existent xfs filesystem mounted at /home
[  171.543553] org.fedoraproject.Anaconda.Modules.Storage[1858]: DEBUG:blivet:             DeviceTree.get_device_by_name: name: VolGroup-LogVol02 ; incomplete: True ; hidden: True ;
[  171.571491] org.fedoraproject.Anaconda.Modules.Storage[1858]: DEBUG:blivet:             DeviceTree.get_device_by_name returned non-existent 1024 MiB lvmlv VolGroup-LogVol02 (133) with non-existent xfs filesystem mounted at /home
[  171.605395] org.fedoraproject.Anaconda.Modules.Storage[1858]: DEBUG:anaconda.modules.storage.devicetree.handler:Mount options of VolGroup-LogVol02 are set to 'nodev,nosuid'.
[  171.633800] org.fedoraproject.Anaconda.Modules.Storage[1858]: DEBUG:blivet:             DeviceTree.get_device_by_name: name: VolGroup-LogVol01 ; incomplete: True ; hidden: True ;
[  171.661737] org.fedoraproject.Anaconda.Modules.Storage[1858]: DEBUG:blivet:             DeviceTree.get_device_by_name returned non-existent 1024 MiB lvmlv VolGroup-LogVol01 (142) with non-existent xfs filesystem mounted at /tmp
[  171.695060] org.fedoraproject.Anaconda.Modules.Storage[1858]: DEBUG:blivet:             DeviceTree.get_device_by_name: name: VolGroup-LogVol01 ; incomplete: True ; hidden: True ;
[  171.721793] org.fedoraproject.Anaconda.Modules.Storage[1858]: DEBUG:blivet:             DeviceTree.get_device_by_name returned non-existent 1024 MiB lvmlv VolGroup-LogVol01 (142) with non-existent xfs filesystem mounted at /tmp
[  171.755214] org.fedoraproject.Anaconda.Modules.Storage[1858]: DEBUG:anaconda.modules.storage.devicetree.handler:Mount options of VolGroup-LogVol01 are set to 'nodev,noexec,nosuid'.
[  171.783773] org.fedoraproject.Anaconda.Modules.Storage[1858]: DEBUG:blivet:             DeviceTree.get_device_by_name: name: VolGroup-LogVol03 ; incomplete: True ; hidden: True ;
[  171.813837] org.fedoraproject.Anaconda.Modules.Storage[1858]: DEBUG:blivet:             DeviceTree.get_device_by_name returned non-existent 3 GiB lvmlv VolGroup-LogVol03 (160) with non-existent xfs filesystem mounted at /var
[  171.851009] org.fedoraproject.Anaconda.Modules.Storage[1858]: DEBUG:blivet:             DeviceTree.get_device_by_name: name: VolGroup-LogVol03 ; incomplete: True ; hidden: True ;
[  171.879329] org.fedoraproject.Anaconda.Modules.Storage[1858]: DEBUG:blivet:             DeviceTree.get_device_by_name returned non-existent 3 GiB lvmlv VolGroup-LogVol03 (160) with non-existent xfs filesystem mounted at /var
[  171.917820] org.fedoraproject.Anaconda.Modules.Storage[1858]: DEBUG:anaconda.modules.storage.devicetree.handler:Mount options of VolGroup-LogVol03 are set to 'defaults,nodev,nosuid'.
[  171.947796] org.fedoraproject.Anaconda.Modules.Storage[1858]: DEBUG:blivet:             DeviceTree.get_device_by_name: name: VolGroup-LogVol04 ; incomplete: True ; hidden: True ;
[  171.976923] org.fedoraproject.Anaconda.Modules.Storage[1858]: DEBUG:blivet:             DeviceTree.get_device_by_name returned non-existent 1024 MiB lvmlv VolGroup-LogVol04 (169) with non-existent xfs filesystem mounted at /var/log
[  172.025301] org.fedoraproject.Anaconda.Modules.Payloads[1855]: DEBUG:anaconda.modules.payloads.payload.dnf.dnf:Packages are set to 'PackagesConfigurationData(broken_ignored=False, core_group_enabled=True, default_environment_enabled=False, docs_excluded=False, environment='graphical-server-environment', excluded_groups=[], excluded_packages=['openldap-clients', 'rsync-daemon', 'net-snmp', 'dhcp-server', 'mcstrans', 'tftp-server', 'tftp', 'avahi-autoipd', 'bind', 'telnet-server', 'squid', 'avahi', 'dovecot', 'vsftpd', 'samba', 'httpd', 'telnet'], groups=['Base'], groups_package_types={}, languages='all', missing_ignored=False, multilib_policy='best', packages=['python3', 'scap-security-guide', 'audit', 'aide', 'nftables', 'sudo', 'libselinux', 'rsyslog'], retries=-1, timeout=-1, weakdeps_excluded=False)'.
[  172.191747] org.fedoraproject.Anaconda.Modules.Network[1873]: DEBUG:anaconda.modules.network.firewall.firewall:Services that will be allowed through the firewall: ['ssh']

....[snip]...

[  175.294549] anaconda[1803]: anaconda: ui.gui.helpers: kickstart installation stopped for info: User interaction required on spoke Security Profile
[  175.296783] org.fedoraproject.Anaconda.Modules.Storage[1858]: DEBUG:blivet:             DeviceTree.get_device_by_name returned non-existent 1024 MiB lvmlv VolGroup-LogVol04 (169) with non-existent xfs filesystem mounted at /var/log
[  175.300897] org.fedoraproject.Anaconda.Modules.Storage[1858]: DEBUG:blivet:             DeviceTree.get_device_by_name: name: VolGroup-LogVol05 ; incomplete: True ; hidden: True ;
[  175.304364] org.fedoraproject.Anaconda.Modules.Storage[1858]: DEBUG:blivet:             DeviceTree.get_device_by_name returned non-existent 512 MiB lvmlv VolGroup-LogVol05 (178) with non-existent xfs filesystem mounted at /var/log/audit
[  175.308054] org.fedoraproject.Anaconda.Modules.Storage[1858]: DEBUG:blivet:             DeviceTree.get_device_by_name: name: VolGroup-LogVol7 ; incomplete: True ; hidden: True ;
[  175.311489] org.fedoraproject.Anaconda.Modules.Storage[1858]: DEBUG:blivet:             DeviceTree.get_device_by_name returned non-existent 1024 MiB lvmlv VolGroup-LogVol7 (151) with non-existent xfs filesystem mounted at /var/tmp
[  175.314553] org.fedoraproject.Anaconda.Modules.Storage[1858]: DEBUG:blivet:             DeviceTree.get_device_by_name: name: VolGroup-LogVol02 ; incomplete: True ; hidden: True ;
[  175.315181] org.fedoraproject.Anaconda.Modules.Storage[1858]: DEBUG:blivet:             DeviceTree.get_device_by_name returned non-existent 1024 MiB lvmlv VolGroup-LogVol02 (133) with non-existent xfs filesystem mounted at /home
[  175.318864] org.fedoraproject.Anaconda.Modules.Storage[1858]: DEBUG:blivet:             DeviceTree.get_device_by_name: name: VolGroup-LogVol01 ; incomplete: True ; hidden: True ;
[  175.324352] org.fedoraproject.Anaconda.Modules.Storage[1858]: DEBUG:blivet:             DeviceTree.get_device_by_name returned non-existent 1024 MiB lvmlv VolGroup-LogVol01 (142) with non-existent xfs filesystem mounted at /tmp
[  175.325520] org.fedoraproject.Anaconda.Modules.Storage[1858]: DEBUG:blivet:             DeviceTree.get_device_by_name: name: VolGroup-LogVol03 ; incomplete: True ; hidden: True ;
[  175.330503] org.fedoraproject.Anaconda.Modules.Storage[1858]: DEBUG:blivet:             DeviceTree.get_device_by_name returned non-existent 3 GiB lvmlv VolGroup-LogVol03 (160) with non-existent xfs filesystem mounted at /var
[  175.336244] org.fedoraproject.Anaconda.Modules.Storage[1858]: DEBUG:blivet:             DeviceTree.get_device_by_name: name: VolGroup-LogVol04 ; incomplete: True ; hidden: True ;
[  175.340263] org.fedoraproject.Anaconda.Modules.Storage[1858]: DEBUG:blivet:             DeviceTree.get_device_by_name returned non-existent 1024 MiB lvmlv VolGroup-LogVol04 (169) with non-existent xfs filesystem mounted at /var/log
[  175.344234] org.fedoraproject.Anaconda.Modules.Storage[1858]: DEBUG:blivet:             DeviceTree.get_device_by_name: name: VolGroup-LogVol05 ; incomplete: True ; hidden: True ;
[  175.351167] systemd[1]: systemd-hostnamed.service: Deactivated successfully.
[  175.364008] org.fedoraproject.Anaconda.Modules.Storage[1858]: DEBUG:blivet:             DeviceTree.get_device_by_name returned non-existent 512 MiB lvmlv VolGroup-LogVol05 (178) with non-existent xfs filesystem mounted at /var/log/audit
[  175.365087] org.fedoraproject.Anaconda.Modules.Storage[1858]: DEBUG:blivet:             DeviceTree.get_device_by_name: name: VolGroup-LogVol7 ; incomplete: True ; hidden: True ;
[  175.370264] org.fedoraproject.Anaconda.Modules.Storage[1858]: DEBUG:blivet:             DeviceTree.get_device_by_name returned non-existent 1024 MiB lvmlv VolGroup-LogVol7 (151) with non-existent xfs filesystem mounted at /var/tmp
[  175.377168] anaconda[1803]: anaconda: ui.gui.helpers: kickstart installation stopped for info: User interaction required on spoke Security Profile
[  904.628512] systemd[1]: Starting Cleanup of Temporary Directories...

```

#### Expected Results:
Install with kickstart finishes

### PR DESCRIPTION
**Pull request by Mab879**
#### Description:

Since gnome-desktop requires libsane-hpaio which requires avahi having avahi removed during the install of the OS causes the install to fail.

#### Rationale:
Fixes #10277

### CODE REVIEW

**Review by mildas — COMMENTED** [review]
Why to disable `package_avahi_removed` when it doesn't conflict with GUI packages?

Looking at RHEL9 CIS Benchmark, I don't see `avahi-autoipd` mentioned at all. `2.2.2 Ensure Avahi Server is not installed (Automated)` requires only `avahi` uninstalled. Why do we remove `avahi-autoipd` package?

**Comment by Mab879** [comment]
> Why to disable `package_avahi_removed` when it doesn't conflict with GUI packages?

Without creating another rule very similar rule I don't see how we can disable this remediation for only one profile.

> Looking at RHEL9 CIS Benchmark, I don't see `avahi-autoipd` mentioned at all. `2.2.2 Ensure Avahi Server is not installed (Automated)` requires only `avahi` uninstalled. Why do we remove `avahi-autoipd` package?

I have removed `avahi-autoipd` from the profile and the package doesn't seem to be in RHEL 9.

File: linux_os/guide/services/avahi/disable_avahi_group/package_avahi-autoipd_removed/rule.yml [hunk 1]
```diff
@@ -45,3 +45,5 @@ template:
     name: package_removed
     vars:
         pkgname: avahi-autoipd
+    backends:
+      anaconda: 'off'
```
Comment (mildas) [inline linux_os/guide/services/avahi/disable_avahi_group/package_avahi-autoipd_removed/rule.yml:49]: those lines can be removed as it's not problematic

**Comment by mildas** [comment]
> Without creating another rule very similar rule I don't see how we can disable this remediation for only one profile.

I see. But I don't like disabling Anaconda remediation. Workstation users won't pass there and so we should think if the rule makes sense in CIS Workstation profile.
Do you think we could convince CIS that the requirement should be removed from Workstation L2 because of the conflict on RHEL9? Or if they could change the requirement to `Avahi not installed or avahi-daemon disabled`?

**Comment by Mab879** [comment]
> > Without creating another rule very similar rule I don't see how we can disable this remediation for only one profile.
> 
> I see. But I don't like disabling Anaconda remediation. Workstation users won't pass there and so we should think if the rule makes sense in CIS Workstation profile. Do you think we could convince CIS that the requirement should be removed from Workstation L2 because of the conflict on RHEL9? Or if they could change the requirement to `Avahi not installed or avahi-daemon disabled`?

I don't know if we could convince them to pull it its in L2, so some pain is expected. Moving to disabled might work, but again a level 2 requirement is expected to have some usability issues.

The package can be removed without incident once the installation is done. Which is why I pulled the Anaconda remediation.

A third option would be get `libsane-hpaio` a soft requirement. 

My thought is that for now I will pull the rule from level II workstation profile as work on the other solution.

**Review by mildas — APPROVED** [review]

**Pull request merged**